modules/react/popup/lib/hooks/useReturnFocus.tsx

```diff
@@ -30,19 +57,62 @@ export const useReturnFocus = createElemPropsHook(usePopupModel)(model => {
     requiresFocusChangeRef.current = false;
   }, []);
 
+  // track mousedown events to determine if the mouse target is a focusable element. If it is, we
+  // should not return focus
+  const onMouseDown = React.useCallback((event: MouseEvent) => {
+    elementRef.current = event.target as Element;
+  }, []);
+
+  const onMouseUp = React.useCallback(() => {
+    elementRef.current = null;
+  }, []);
+
+  // We use `useLayoutEffect` because the callback will be called _before_ the browser changes
+  // focus. This allows the browser to change focus as normal. For example, if the popup closes
+  // because the user hits the tab key when focused on the last focusable element,
+  // `useFocusRedirect` will cause the popup to close before the browser is finished processing the
+  // `keydown`. The browser's default action of a keydown of a tab key is to advance focus to the
+  // next focusable element. This means that the `useFocusRedirect` will change focus back to the
+  // target, then the tab key is processed by the browser to focus on the next item after the target
+  // without us having to use a focus manager to guess what the next focusable element is.
+  //
+  // But because we're using `useLayoutEffect` instead of `useEffect`, we cannot use
+  // `document.activeElement` (because the focus hasn't changed yet because the browser hasn't
+  // triggered a default action yet) to determine if a user clicked on something that can accept
+  // focus itself. If we don't short-circuit return focus, the user will intend to click on an
+  // `input` element only to get focus changed back. This is annoying to users and unintuitive. Instead,
+  // we'll use `mousedown` and `mouseup` to determine if we're closing because of a mouse click.
```

Comment:
Do we have to account for `blur` events or does mouse down and up cover our cases?